src/helpers/figures/charts/scorecard_chart.ts

```diff
@@ -177,6 +179,102 @@ export class ScorecardChart extends AbstractChart {
   }
 }
 
+function drawDecoratedText(
+  context: CanvasRenderingContext2D,
+  text: string,
+  position: PixelPosition,
+  underline: boolean | undefined = false,
+  strikethrough: boolean | undefined = false
+) {
+  context.fillText(text, position.x, position.y);
+  const measure = context.measureText(text);
+  const textWidth = measure.width;
+  const textHeight = measure.actualBoundingBoxAscent + measure.actualBoundingBoxDescent;
+  const lineWidth = Number(context.font.match(/([0-9\.]*)px/)?.[1]) / 10;
+  if (underline) {
+    context.lineWidth = lineWidth;
+    context.strokeStyle = context.fillStyle;
+    context.beginPath();
+    context.moveTo(position.x, position.y + lineWidth);
+    context.lineTo(position.x + textWidth, position.y + lineWidth);
+    context.stroke();
+  }
+  if (strikethrough) {
+    context.lineWidth = lineWidth;
+    context.strokeStyle = context.fillStyle;
+    context.beginPath();
+    context.moveTo(position.x, position.y - textHeight / 2 + lineWidth);
+    context.lineTo(position.x + textWidth, position.y - textHeight / 2 + lineWidth);
+    context.stroke();
+  }
+}
+
+export function drawScoreChart(structure: ScorecardChartDesign, canvas: HTMLCanvasElement) {
+  // Set canvas size
+  const ctx = canvas.getContext("2d")!;
+  canvas.width = structure.canvas.width;
```

Comment:
Now that we have the canvas that will take the full width/height of the container, I would like to get rid of the figure props altogether since it's only present for this specific type of charts.
Here we can infer the element size `ctx.canvas.getBoundingClientRect`.
So ctx could be passed as an argument to bother `drawScoreChart` and `ScorecardChartDesign`.
@LucasLefevre wdyt?

Reply:
> EDIT: image needs the figure id to get the image url :/
I meant for figureChart specifically ;-)

Reply:
Bump on this one - we have to pass the chartId (resp. imageId) to `ChartFigure` (resp. ImageFigure) but that's it. the figure object is not required. I really think we can get rid of it and use the canvas as the container reference to compute the scorecard  config.